src/app/active-mode/Views/current-activity/VendorBounties.tsx

```diff
@@ -1,110 +1,179 @@
+
+    return matchingDestination || matchingActivity;
+  });
+}
 
-/** Find unclaimed vendor bounties based on your current activity */
+/** Find relevant vendor bounties based on your current activity */
 function VendorBounties({
   defs,
-  vendors,
+  vendorsResponse,
   store,
-  activity,
+  activityInfo,
   buckets,
   ownedItemHashes,
   account,
 }: Props) {
-  if (!vendors) {
-    return null;
-  }
-
-  const bounties: VendorItem[] = [];
-  const vendorData = store.id ? vendors[store.id] : undefined;
-  const vendorsResponse = vendorData?.vendorsResponse;
-  const vendorHashes: Vendors[] = [];
-  activity.activityModeTypes?.forEach((modeType) => {
-    const vendors = vendorsByActivityModeType[modeType];
-    vendors && vendorHashes.push(...vendors);
-  });
+  const [suggestedBounties, setSuggestedBounties] = useState<VendorItem[]>([]);
+  const [bounties, setBounties] = useState<VendorItem[]>([]);
 
-  if (!vendorHashes.length) {
-    const vendors = vendorsByDestinationHash[activity.placeHash];
-    vendors && vendorHashes.push(...vendors);
-  }
-
-  vendorHashes.forEach((vendorHash) => {
-    const vendor = vendorsResponse?.vendors.data?.[vendorHash];
-    const d2Vendor = toVendor(
-      vendorHash,
-      defs,
-      buckets,
-      vendor,
-      account,
-      vendorsResponse?.itemComponents[vendorHash],
-      vendorsResponse?.sales.data?.[vendorHash]?.saleItems,
-      {}
-    );
-    const vendorBounties = d2Vendor?.items.filter(
-      ({ item, canPurchase, canBeSold }: VendorItem) =>
-        canPurchase && canBeSold && item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties)
-    );
+  useEffect(() => {
+    if (!vendorsResponse) {
+      return;
+    }
+
+    const purchasableBounties: VendorItem[] = [];
+    const { vendors, sales, itemComponents } = vendorsResponse;
+
+    Object.values(vendors.data || []).forEach((vendor) => {
+      const { vendorHash } = vendor;
+      const d2Vendor = toVendor(
+        vendorHash,
+        defs,
+        buckets,
+        vendor,
+        account,
+        itemComponents[vendorHash],
+        sales.data?.[vendorHash]?.saleItems,
+        {}
+      );
+
+      d2Vendor &&
+        purchasableBounties.push(
+          ...d2Vendor.items.filter(
+            ({ item, canPurchase, canBeSold }: VendorItem) =>
+              canPurchase &&
+              canBeSold &&
+              item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties)
+          )
+        );
+    });
+
+    setBounties(purchasableBounties);
+  }, [defs, account, buckets, vendorsResponse]);
 
-    if (vendorBounties) {
-      bounties.push(...vendorBounties);
+  useEffect(() => {
+    setSuggestedBounties(bountiesForActivity(defs, bounties, activityInfo));
+  }, [defs, bounties, activityInfo]);
+
+  const ownedBountyHashes: number[] = [];
+  const unownedBounties: VendorItem[] = [];
+  suggestedBounties.forEach((vendorItem) => {
```

Comment:
Totally fine but FYI https://lodash.com/docs/4.17.15#partition

Reply:
ah, weird bit here is that one is a new array of numbers and the other is a list of vendor items. I was looking for a function like that partition one though, going to keep that in my back pocket